processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -117,6 +121,33 @@ public ExtractionFn getExtractionFn()
   @Override
   public DimFilter optimize()
   {
+    if (values.size() == 1) {
+      return new SelectorDimFilter(dimension, values.first(), extractionFn).optimize();
+    }
+    // Similar optimization as SelectorDimFilter
+    if (extractionFn instanceof LookupExtractionFn
+            && ((LookupExtractionFn) extractionFn).isOptimize()) {
+      LookupExtractionFn exFn = (LookupExtractionFn) extractionFn;
+      LookupExtractor lookup = exFn.getLookup();
+
+      final List<String> keys = new ArrayList<>();
+      for (String value : values) {
+        final String convertedValue = Strings.emptyToNull(value);
+        if (!exFn.isRetainMissingValue() && Objects.equals(convertedValue, exFn.getReplaceMissingValueWith())) {
+          return this;
+        }
+        keys.addAll(lookup.unapply(convertedValue));
+        if (exFn.isRetainMissingValue() && lookup.apply(convertedValue) == null) {
+          keys.add(convertedValue);
+        }
+      }
+
+      if (keys.isEmpty()) {
+        return this;
+      } else {
+        return new InDimFilter(dimension, keys, null).optimize();
+      }
+    }
```

Comment:
We could put all of the optimization code in `InDimFilter` and have `SelectorDimFilter.optimize` do `new InDimFilter(dimension, ImmutableSet.of(value), extractionFn).optimize()`